lib/ivis_opengl/imdload.cpp

```diff
@@ -616,11 +617,11 @@ bool _imd_load_connectors(const char **ppFileData, iIMDShape *s)
 }
 
 // performance hack
-static QVector<GLfloat> vertices;
-static QVector<GLfloat> normals;
-static QVector<GLfloat> texcoords;
-static QVector<GLfloat> tangents;
-static QVector<uint16_t> indices; // size is npolys * 3 * numFrames
+static std::vector<float> vertices;
+static std::vector<float> normals;
+static std::vector<float> texcoords;
+static std::vector<float> tangents;
```

Comment:
Replacing `GL` types with C types is not a good idea.

According to Table 2.2 of the [OpenGL spec](https://www.khronos.org/registry/OpenGL/specs/gl/glspec40.core.pdf):
> GL types are not C types. Thus, for example, GL type `int` is referred to as `GLint` outside this document, and **is not necessarily equivalent to the C type `int`**. An implementation may use more bits than the number indicated in the table to represent a GL type.

(Emphasis mine.)

By doing this, we are reducing the cross-platform compatibility of the code, and increasing the likelihood of a mismatch with a platform's specific `gl.h` implementation. If it works today, that's *by accident*, not guaranteed.

If interacting with OpenGL, the GL types ought to be used.

Reply:
The goal is to make graphic buffer API agnostic so that they can be passed to either vulkan or GL.
So far I know no GL implementation that deviates from the usual type size (ie float as 4 bytes, int as 4 bytes, short as 2 bytes...) : Intel HD, AMD and Nvidia does that on Windows Linux x86 ARM and likely PPC (it's rather hard to find affordable PPC hw these days), Mali and Qualcomm do that on mobile... Since GPU are designed to be feed by buffer they have to agree on some convention. Some ASIC uses 40 bits int but their often used in embedded hw for audio processing and doesn't support GL.
Vulkan doesn't even redefine these types either. 

The only exception I see is GLBoolean, it's often implemented as int by gpu to follow the 4 bytes alignment rule.

Reply:
I understand what your goal is, but you can simply use a typedef in your gfx_api wrapper to encapsulate this. For GL backend builds, typedef the new "gfxFloat" (or whatever you name it) to `GLfloat`.